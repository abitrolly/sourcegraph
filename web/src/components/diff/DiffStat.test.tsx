import React from 'react'
import { DiffStat } from './DiffStat'
import { mount } from 'enzyme'

describe('DiffStat', () => {
    test('standard', () => expect(mount(<DiffStat added={1} changed={2} deleted={3} />)).toMatchSnapshot())

    test('expandedCounts', () =>
        expect(mount(<DiffStat added={1} changed={2} deleted={3} expandedCounts={true} />)).toMatchSnapshot())
})
