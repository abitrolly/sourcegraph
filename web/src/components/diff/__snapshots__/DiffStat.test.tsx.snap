// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffStat expandedCounts 1`] = `
<DiffStat
  added={1}
  changed={2}
  deleted={3}
  expandedCounts={true}
>
  <div
    className="diff-stat "
    data-tooltip="1 addition, 2 changes, 3 deletions"
  >
    <span
      className="diff-stat__total font-weight-bold"
    >
      <span
        className="diff-stat__text-added mr-1"
      >
        +
        1
      </span>
      <span
        className="diff-stat__text-changed mr-1"
      >
        •
        2
      </span>
      <span
        className="diff-stat__text-deleted mr-1"
      >
        −
        3
      </span>
    </span>
    <div
      className="diff-stat__square diff-stat__added"
      key="0"
    />
    <div
      className="diff-stat__square diff-stat__changed"
      key="1"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="2"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="3"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="4"
    />
  </div>
</DiffStat>
`;

exports[`DiffStat standard 1`] = `
<DiffStat
  added={1}
  changed={2}
  deleted={3}
>
  <div
    className="diff-stat "
    data-tooltip="1 addition, 2 changes, 3 deletions"
  >
    <small
      className="diff-stat__total"
    >
      8
    </small>
    <div
      className="diff-stat__square diff-stat__added"
      key="0"
    />
    <div
      className="diff-stat__square diff-stat__changed"
      key="1"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="2"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="3"
    />
    <div
      className="diff-stat__square diff-stat__deleted"
      key="4"
    />
  </div>
</DiffStat>
`;
